ponder.config.ts

```diff
@@ -1,29 +1,32 @@
-import { ACTIVE_PLUGIN } from "./src/lib/plugin-helpers";
+import { deepMergeRecursive } from "./src/lib/helpers";
+import { type IntersectionOf, getActivePlugins } from "./src/lib/plugin-helpers";
 import * as baseEthPlugin from "./src/plugins/base.eth/ponder.config";
 import * as ethPlugin from "./src/plugins/eth/ponder.config";
 import * as lineaEthPlugin from "./src/plugins/linea.eth/ponder.config";
 
+// list of all available plugins
+// any of them can be activated in the runtime
 const plugins = [baseEthPlugin, ethPlugin, lineaEthPlugin] as const;
 
-// here we export only a single 'plugin's config, by type it as every config
-// this makes all of the mapping types happy at typecheck-time, but only the
-// relevant config is run at runtime
-export default ((): AllConfigs => {
-  const pluginToActivate = plugins.find((p) => p.ownedName === ACTIVE_PLUGIN);
+// intersection of all available plugin configs to support correct typechecking
+// of the indexing handlers
+type AllPluginsConfig = IntersectionOf<(typeof plugins)[number]["config"]>;
 
-  if (!pluginToActivate) {
-    throw new Error(`Unsupported ACTIVE_PLUGIN: ${ACTIVE_PLUGIN}`);
-  }
+// Activates the indexing handlers included in selected active plugins and
+// returns and intersection of their combined config.
+function getActivePluginsConfig(): AllPluginsConfig {
```

Comment:
@lightwalker-eth updated the comments to express the active plugins config idea more clearly.